src/Likvido.ApplicationInsights.Telemetry/HttpResponse/HttpResponseTelemetryInitializer.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+
+namespace Likvido.ApplicationInsights.Telemetry
+{
+    /// <summary>
+    /// Enrich telemetry with response content.
+    /// </summary>
+    public class HttpResponseTelemetryInitializer : ITelemetryInitializer
+    {
+        private readonly Func<HttpResponseTelemetryConditionInfo, bool> _condition;
+
+        public List<string> ContentMediaTypes { get; } = new List<string>
+        {
+            "application/json",
+            "text/plain",
+            "text/html"
+        };
+
+        public HttpResponseTelemetryInitializer(
+            Func<HttpResponseTelemetryConditionInfo, bool> condition)
+        {
+            _condition = condition ?? throw new ArgumentNullException(nameof(condition));
+        }
+
+        public HttpResponseTelemetryInitializer(
+            Func<HttpResponseTelemetryConditionInfo, bool> condition,
+            List<string> contentMediaTypes) : this(condition)
```

Comment:
Let's do this to make it more flexible
```suggestion
            Func<string, bool> contentMediaTypesCheck) : this(condition)
```